### PR TITLE
Prevent next paste on middle clic with linux

### LINKED ISSUE #1383: Moving with middle button paste unwanted elements on Linux
On linux, if something has been copied beforehand and the screen is moved with the middle mouse button, the copied element is pasted. (The middle button on linux can paste something).

If the mouse has moved, the pasted item should not be taken into account.

Peek 11-04-2020 21-36

### PR DESCRIPTION
**Pull request by jrmi**
Proposal to fix #1383.

I propose to fix the extra middle click paste event only on linux by preventing the next paste event only when user pan the view.

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -1611,6 +1615,24 @@ class App extends React.Component<any, AppState> {
...
+          !nextPastePrevented &&
+          (Math.abs(deltaX) > 1 || Math.abs(deltaY) > 1)
+        ) {
+          nextPastePrevented = true;
+          const preventNextPaste = (e: ClipboardEvent) => {
+            document.body.removeEventListener("paste", preventNextPaste);
+            e.stopPropagation();
+          };
+
+          document.body.addEventListener("paste", preventNextPaste);
```
Comment (lipis) [inline src/components/App.tsx:1633]: https://github.com/excalidraw/excalidraw/blob/master/src/constants.ts#L27

File: src/components/App.tsx [hunk 2]
```diff
@@ -1611,6 +1615,24 @@ class App extends React.Component<any, AppState> {
...
+         * Prevent paste event if we move while middle clicking on Linux.
+         * See issue #1383.
+         */
+        if (
+          isLinux &&
+          !nextPastePrevented &&
+          (Math.abs(deltaX) > 1 || Math.abs(deltaY) > 1)
+        ) {
+          nextPastePrevented = true;
+          const preventNextPaste = (e: ClipboardEvent) => {
```
Comment (lipis) [inline src/components/App.tsx:1628]: `event` instead of `e`

File: src/components/App.tsx (hunk 1 above)
Comment (jrmi) [reply src/components/App.tsx]: FIxed

File: src/components/App.tsx (hunk 2 above)
Comment (jrmi) [reply src/components/App.tsx]: Fixed

**Comment by dwelle** [comment]
What happens on linuxes where rightclick-paste isn't enabled? Right now it'd always prevent first paste after someone wheel-panned.

**Comment by jrmi** [comment]
What do you think of removing the "event blocker" with a 300ms timeout after the `pointerup` event ?

**Comment by dwelle** [comment]
That feels unnecessarily too much. Does 100ms work?

**Comment by jrmi** [comment]
Think so. I do that.

**Comment by jrmi** [comment]
PR updated.

**Comment by hugows** [comment]
I'm having the same problem and its driving me crazy.

**Pull request merged**